### PR TITLE
Cbr

### PR DESCRIPTION
**Pull request by rajatagarwal-ibm**
### Description

* Added CBR capability to the complete example
* no more skipping upgrade tests

### Types of changes in this PR 
https://github.ibm.com/GoldenEye/issues/issues/3766

#### No release required

- [x] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [x] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Replace this text with information that users need to know about the bug fixes, features, and breaking changes. This information helps the merger write the commit message that is published in the release notes for the module.

---

### Checklist for reviewers

- [x] The PR references a GitHub issue.
- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

**Comment by ocofaigh** [comment]
@daniel-butler-irl can you please review and ensure approach is consistent with other ICD modules?

**Review by daniel-butler-irl — CHANGES_REQUESTED** [review]
Looks good just minor update

**Review by daniel-butler-irl — APPROVED** [review]

File: variables.tf [hunk 1]
```diff
@@ -180,3 +180,31 @@ variable "auto_scaling" {
...
+      attributes = optional(list(object({
+        name  = string
+        value = string
+    }))) }))
+    enforcement_mode = string
+    tags = optional(list(object({
+      name  = string
+      value = string
+    })))
+    operations = optional(list(object({
```
Comment (daniel-butler-irl) [inline variables.tf:201]: There is no option for operations so hardcode It to the default value you have above

File: examples/complete/main.tf [hunk 2]
```diff
@@ -74,7 +111,36 @@ module "mongodb" {
...
+  endpoints           = "private"
   region              = var.region
   key_protect_key_crn = module.key_protect_all_inclusive.keys["icd.${var.prefix}-mongodb"].crn
   tags                = var.resource_tags
+  cbr_rules = [
+    {
+      description      = "sample rule"
+      enforcement_mode = "enabled"
+      account_id       = data.ibm_iam_account_settings.iam_account_settings.account_id
+      tags = [
```
Comment (daniel-butler-irl) [inline examples/complete/main.tf:123]: please see cos module for updates around tags

File: examples/complete/variables.tf [hunk 3]
```diff
@@ -34,14 +34,14 @@ variable "mongodb_version" {
   default     = null
 }
 
-variable "sm_service_plan" {
-  type        = string
-  description = "Secrets Manager plan"
-  default     = "trial"
-}
-
 variable "service_credentials" {
```
Comment (daniel-butler-irl) [inline examples/complete/variables.tf:37]: don't need these anymore, they where part of the SM flow we removed

File: main.tf [hunk 4]
```diff
@@ -85,3 +85,48 @@ resource "ibm_database" "mongodb" {
...
+
+##############################################################################
+# Context Based Restrictions
+##############################################################################
+
+module "cbr_rule" {
+  count            = length(var.cbr_rules) > 0 ? length(var.cbr_rules) : 0
+  source           = "git::https://github.com/terraform-ibm-modules/terraform-ibm-cbr//cbr-rule-module?ref=v1.1.2"
+  rule_description = var.cbr_rules[count.index].description
+  enforcement_mode = var.cbr_rules[count.index].enforcement_mode
```
Comment (daniel-butler-irl) [inline main.tf:97]: this reference is incorrect, as it is currently implemented enforcement mode is not in the cbr_rules variable. Should be just `var.enforcement_mode`

File: examples/complete/main.tf [hunk 5]
```diff
@@ -74,7 +94,26 @@ module "mongodb" {
   resource_group_id   = module.resource_group.resource_group_id
```
Comment (daniel-butler-irl) [inline examples/complete/main.tf:94]: Need to add a dependency. You won't see the failure in the test but if you manually apply you will probably hit it.
``` hcl
  // Policy needs to be applied before CRB is created or there will be a failure
  depends_on = [ibm_iam_authorization_policy.policy]
```

Comment (rajatagarwal-ibm) [reply examples/complete/main.tf:94]: Actually, just ran it manually and it passed.

Comment (daniel-butler-irl) [reply examples/complete/main.tf:94]: You are right not needed

File: examples/complete/main.tf [hunk 6]
```diff
@@ -56,7 +94,26 @@ module "mongodb" {
...
   mongodb_version     = var.mongodb_version
   instance_name       = "${var.prefix}-mongodb"
+  endpoints           = "private"
   region              = var.region
   key_protect_key_crn = module.key_protect_all_inclusive.keys["icd.${var.prefix}-mongodb"].crn
   tags                = var.resource_tags
+  cbr_rules = [
+    {
+      description      = "sample rule"
```
Comment (daniel-butler-irl) [inline examples/complete/main.tf:103]: In the UI this is the main identifier so update it to something like `${var.prefix}-mongodb access only from vpc`

File: main.tf [hunk 7]
```diff
@@ -85,3 +85,48 @@ resource "ibm_database" "mongodb" {
...
+        value    = ibm_database.mongodb.guid
+        operator = "stringEquals"
+      },
+      {
+        name     = "serviceName"
+        value    = "databases-for-mongodb"
+        operator = "stringEquals"
+      }
+    ],
+    tags = var.cbr_rules[count.index].tags != null ? var.cbr_rules[count.index].tags : [
```
Comment (daniel-butler-irl) [inline main.tf:117]: default tag needs to be removed

File: tests/pr_test.go [hunk 8]
```diff
@@ -79,15 +80,15 @@ func TestRunCompleteExample(t *testing.T) {
...
-	t.Skip("Skipping upgrade test until initial code is in primary branch")
-
 	options := testhelper.TestOptionsDefaultWithVars(&testhelper.TestOptions{
 		Testing:            t,
-		TerraformDir:       defaultExampleTerraformDir,
+		TerraformDir:       completeExampleTerraformDir,
 		Prefix:             "mongodb-upg",
 		ResourceGroup:      resourceGroup,
 		BestRegionYAMLPath: regionSelectionPath,
+		TerraformVars: map[string]interface{}{
```
Comment (daniel-butler-irl) [inline tests/pr_test.go:89]: after some more testing, this is not needed.

File: examples/complete/README.md [hunk 9]
```diff
@@ -1,4 +1,4 @@
-# Encryption example
+# Complete example with Encryption and CBR rules
```
Comment (daniel-butler-irl) [inline examples/complete/README.md:1]: the rest of this readme is out of date and does not reflect what is in the module

File: main.tf [hunk 10]
```diff
@@ -85,3 +85,43 @@ resource "ibm_database" "mongodb" {
...
+        value    = ibm_database.mongodb.guid
+        operator = "stringEquals"
+      },
+      {
+        name     = "serviceName"
+        value    = "databases-for-mongodb"
+        operator = "stringEquals"
+      }
+    ],
+    tags = var.cbr_rules[count.index].tags
```
Comment (daniel-butler-irl) [inline main.tf:117]: From yesterday's deep dive, it was decided we would remove CBR tags altogether as they have no value here

**Review by daniel-butler-irl — APPROVED** [review]

**Pull request merged**